src/common/time/src/timestamp.rs

```diff
@@ -143,7 +143,8 @@ impl Timestamp {
             Some(Timestamp::new(value, unit))
         } else {
             let mul = unit.factor() / self.unit().factor();
-            Some(Timestamp::new(self.value.div_ceil(mul as i64), unit))
+            let new_ts = self.value as f64 / mul as f64;
```

Comment:
I'm afraid this conversion will lose some precision. For example, let `mul` = 1000, `self.value` = 9223372036854676001 (a random number big enough),  `self.value.div_ceil(mul)` gives you 9223372036854677 while `(self.value as f64 / mul as f64).ceil() as i64` gives you 9223372036854676, which is incorrect.

Reply:
Maybe we can port the `div_ceil` function:
```rs
#[inline]
pub const fn div_ceil(self, rhs: Self) -> Self {
    let d = self / rhs;
    let r = self % rhs;
    if (r > 0 && rhs > 0) || (r < 0 && rhs < 0) {
        d + 1
    } else {
        d
    }
}
```

Reply:
Confirmed. It happens when `self.value` is big enough for 64-bit value.